ludwig/utils/torch_utils.py

```diff
@@ -93,6 +93,23 @@ def losses(self):
             collected_losses.extend(layer.losses)
         return collected_losses
 
+
+class LudwigComponent(Module):
+    def __init__(self):
+        super().__init__()
+
+    @property
+    def input_shape(self) -> torch.Size:
+        """ Returns the size of the input tensor without the batch dimension. """
+        raise NotImplementedError('Abstract class.')
+
+    @property
+    def output_shape(self) -> torch.Size:
+        """ Returns the size of the output tensor without the batch dimension."""
```

Comment:
The input shape will not include batch size, so we should account for that. PyTorch Summary uses batch size 2 for batch norm. Here's an example:

https://github.com/sksq96/pytorch-summary/blob/master/torchsummary/torchsummary.py#L59

Reply:
Good catch, I've added a dummy batch dimension of 2 right now!